intellij/src/saros/intellij/eventhandler/filesystem/LocalFilesystemModificationHandler.java

```diff
@@ -369,7 +388,13 @@ private void generateResourceDeletionActivity(@NotNull VirtualFileEvent virtualF
 
     SPath path = VirtualFileConverter.convertToSPath(project, deletedVirtualFile);
 
-    if (path == null || !session.isShared(path.getResource())) {
+    IResource resource =
+        intelliJReferencePointManager.getSarosResource(
+            path.getReferencePoint(), path.getProjectRelativePath());
+
+    IFile file = resource.adaptTo(IntelliJFileImpl.class);
```

Comment:
This resource does not have to be a file. This casting is unnecessary. Just use the `IResource` reference.